subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/ValidatePlugins.java

```diff
@@ -76,120 +54,56 @@
  */
 @CacheableTask
 @Incubating
-public class ValidatePlugins extends DefaultTask {
+public abstract class ValidatePlugins extends DefaultTask {
     private final ConfigurableFileCollection classes;
     private final ConfigurableFileCollection classpath;
     private final RegularFileProperty outputFile;
     private final Property<Boolean> enableStricterValidation;
     private final Property<Boolean> ignoreFailures;
     private final Property<Boolean> failOnWarning;
 
-    @Inject
-    public ValidatePlugins(ObjectFactory objects) {
-        this.classes = objects.fileCollection();
-        this.classpath = objects.fileCollection();
-        this.outputFile = objects.fileProperty();
-        this.enableStricterValidation = objects.property(Boolean.class).convention(false);
-        this.ignoreFailures = objects.property(Boolean.class).convention(false);
-        this.failOnWarning = objects.property(Boolean.class).convention(true);
+    public ValidatePlugins() {
+        this.classes = getObjects().fileCollection();
```

Comment:
Since you made the class abstract, you could only declare abstract getters and not have fields at all. Then you also wouldn't need to inject `ObjectFactory`.

Reply:
Makes sense.

Reply:
This removed 25 lines of boilerplate, \o/